### PR TITLE
Increment Version to 1.5.0 and add new guidelines to MAINTAINERS_GUIDE.md and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by IanHoang**
### Description
Increment version.txt after release and adding new guidelines for contributors and maintainers.
### Issues Resolved
N/A
### Testing
N/A

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by beaioun — APPROVED** [review]

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:18]: Use the `backport 3` label for every PR, since main should be the image of `3`, at least until OpenSearch version 3 is officially released.  Add other branch labels appropriately; almost certainly `backport 2` will also be needed, since this is the current OpenSearch version.

File: MAINTAINERS_GUIDE.md
Comment (gkamat) [inline MAINTAINERS_GUIDE.md:19]: Close out the backport PRs in chronological order.  If you close out a PR on `main`, please close out the associated branch PRs as well.  They get generated in a few minutes by the automated backport workflow.  Once they are merged in, the backport branches can be deleted as well.

File: CONTRIBUTING.md
Comment (gkamat) [inline CONTRIBUTING.md:93]: We aim to review all contributions promptly but cannot guarantee quick turnaround times. It can take time for appropriate analysis, verification and testing, to understand the implications and ramifications of the proposed change, and for discussion with other maintainers and developers. Furthermore, it is essential that contributions remain accessible for an adequate duration to allow healthy engagement by community members, other than maintainers.

Please note that there is a code freeze imposed starting a week before every release to ensure stability of the codebase and a smooth release process.  If you are interested in getting a change into the next release, please submit your change well in advance, so that it is approved and merged in prior to the deadline for the freeze.  We are in the process of implementing snapshot releases, so if your change does not make a particular release, the appropriate snapshot release can be used until the next official release is cut.

**Review by gkamat — CHANGES_REQUESTED** [review]

**Review by gkamat — APPROVED** [review]

**Pull request merged**
